### ISSUE TITLE
Divergence in case of zeroed Input

### ISSUE THREAD

**Issue by ChipsSpectre** [zylo117/Yet-Another-EfficientDet-Pytorch#277]
Hello zylo,

thank you again for the great repository which is really helpful for my team and me!

Today I found out that the training diverges to **NaN** loss from the 2nd iteration in case a tensor of zeros is used in the following simplistic training scenario. But if I use a tensor of ones the training works smoothly.
What could be the reasons for this behavior?
```
compound_coef = 0
num_classes = 3

anchors_scales = [2 ** 0, 2 ** (1.0 / 3.0), 2 ** (2.0 / 3.0)]
anchors_ratios = [(1.0, 1.0), (1.4, 0.7), (0.7, 1.4)]

model = EfficientDetBackbone(num_classes=num_classes, compound_coef=compound_coef,
                                ratios=anchors_ratios, scales=anchors_scales).cuda()
model = ModelWithLoss(model)
tensor = torch.ones([1, 3, 256, 256]).cuda()
annot = torch.FloatTensor([0, 0, 100, 100, 1]).reshape([1, 1, 5]).cuda()

lr = 1e-10
optimizer = torch.optim.SGD(model.parameters(), lr, momentum=0.9, nesterov=True)
for i in range(5):
    optimizer.zero_grad()
    cls_loss, reg_loss = model(tensor, annot)
    loss = cls_loss.mean() + reg_loss.mean()
    loss.backward()
    optimizer.step()
    print(cls_loss, reg_loss)
```

**Comment by zylo117** [comment]
it might has something to do with weights init.
And there are lots of bn and sigmoid in this model, which make value around 0 be more sensitive.
A tensor full of zeros should maximize the gradients of sigmoid operations, right? So the loss should higher than ever.
If you want to test, you should use randn instead of ones or zeros.
